scripts/Snakefile

```diff
@@ -133,10 +142,45 @@ rule load_dataset:
     output: temp("{tempdir}/{task}/{dataset}.data.h5ad")
     shell: "python3 {input.script} {wildcards.task} {wildcards.dataset} {output}"
 
-rule run_docker:
+rule login_docker:
+    output:
+        temp(".docker_login")
+    params:
+        password=os.environ['DOCKER_PASSWORD']
+    shell:
+        "docker login --username=singlecellopenproblems --password={params.password} && touch {output}"
+
+rule build_docker:
+    input:
+        dockerfile="../docker/base/Dockerfile",
+        login=".docker_login",
+        code = "{}/../openproblems/version.py".format(SCRIPTS_DIR)
+    output:
+        temp(".docker_build")
+    shell:
+        "docker build -f {input.dockerfile} -t singlecellopenproblems/openproblems . && touch {output}"
```

Comment:
Maybe you want to tag with a wildcard here so that we only build a docker image once per method if the method hasn't changed?